File: test/data/typescript/test01/StateState2.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateState2 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public TransitCommand3(): BaseState | null
    {
        this._controlleeImp.DoAction3();
        return null;
    }
    public  GetStateName(): string
    {
        return "State2";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
    public GetStateID(): number
    {
        return 1;
    }
}

File: test/data/cpp/test01/StateState2.cpp
#include "StateState2.h"
StateState2::StateState2(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState2::~StateState2()
{
}
BaseState* StateState2::TransitCommand3()
{
    _controllee->DoAction3();
    return 0;
}
const char* StateState2::GetStateName()
{
    return "State2";
}
BaseState* StateState2::GetParent()
{
    return 0;
}
int StateState2::GetStateID()
{
    return 1;
}